.github/actions/test-fc/action.yml

```diff
@@ -42,13 +43,20 @@ runs:
             args="-L/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib"
           fi
         fi
-
+        
+        # hello world program
         ${{ env.FC }} $args -o hw hw.f90
         output=$(./hw '2>&1')
         [[ "$output" == *"hello world"* ]] && echo "$output" || (echo "Unexpected Fortran program output: $output"; exit 1)
         rm hw
 
+        # program reproducing https://github.com/fortran-lang/setup-fortran/pull/51
+        # expect crash at runtime so don't run, just compile. see link for details
```

Comment:
It won't crash at runtime, that's only if it gets compiled with gcc11 and links against gcc12. We can run it, in fact it will detect if there are similar path issues in the future.